mysensors/sensor.py

```diff
@@ -103,46 +107,92 @@ def add_child_sensor(self, child_id, child_type, description=""):
         self.children[child_id] = ChildSensor(child_id, child_type, description)
         return child_id
 
-    def set_child_value(self, child_id, value_type, value, **kwargs):
-        """Set a child sensor's value."""
-        children = kwargs.get("children", self.children)
-        if not isinstance(children, dict) or child_id not in children:
+    def get_desired_value(self, child_id, value_type):
+        """Return sensor state value taking into account node type."""
+        if child_id not in self.children:
             return None
-        msg_type = kwargs.get("msg_type", 1)
-        ack = kwargs.get("ack", 0)
-        msg = Message().modify(
+
+        value = None
+
+        if self.is_smart_sleep_node:
+            child = self.new_state[child_id]
+            value = child.values.get(value_type) if child else None
+
+        if value:
```

Comment:
I think we need to check for `None` explicitly here since the value may be an empty string, which is valid.

Reply:
@MartinHjelmare I've created a new PR - #406